### DISCUSSION TITLE
Set background for all outputs

### DISCUSSION THREAD

**Discussion by chriselrod** [YaLTeR/niri#617] [Ideas]
More generally, can we add output options independent of the particular output?
```
output "eDP-1" {
    // off
    mode "1920x1080@120.030"
    scale 2.0
    transform "90"
    position x=1280 y=0
    variable-refresh-rate
    background-color "#003300"
}
```
Although some like `position` may not make sense to apply to multiple outputs.

**Comment by YaLTeR** [comment]
I don't think anything else really makes sense independent of output? But background-color yeah. Not sure where to put it though.

**Reply by chriselrod** [reply]
Yeah, none of the others probably make sense.
Maybe allow setting background at toplevel, like screenshot-path and prefer-no-csd?

I thought I might want scale=1 everywhere.
I have a window rule to set foot's max width to 640. Standard monitor widths tend to be a multiple of 640, e.g. 1920 (3*640), 3840 (6*640), or even 5120 (8*640).
The default scale factor might brake this, e.g. it does with my 13' laptop with a 1920-pixel wide screen.

If I had a 4k laptop screen, I'd probably want a scale factor of 2, though.
I think I'd rather have an auto-adjusting scale factor that behaves similarly to the current default, except it restricts the scale factor to picking an integer multiple of a user-configured value (e.g., 640).
That way, a 4k laptop (or otherwise small monitor) could automatically pick a scale factor of 2 or 3, so that 3 or 2 640-wide windows fit, respectively.
Or, maybe allow scaling the window rules widths with the scale factor.
This also mostly keeps the terminal emulators playing nice with windows using proportion-rules.

Or, taking another step back, what I ultimately want is to have an integer number of terminals fit on my monitors. These terminals should generally be narrow, to allow fitting more of them. Proportional rules don't make sense across monitors of vastly different sizes and aspect ratios, e.g. the 1920p 13' laptop vs.
```
$ niri msg outputs
Output "HDMI-A-1" (Unknown - Unknown - HDMI-A-1)
  Current mode: 5120x1440 @ 59.977 Hz (preferred)
  Variable refresh rate: not supported
  Physical size: 1200x340 mm
```
Setting the terminal to 640-wide seemed the most straightforward way to achieve this.
